packages/web-runtime/tests/unit/components/Avatar.spec.js

```diff
@@ -0,0 +1,94 @@
+import { shallowMount, createLocalVue } from '@vue/test-utils'
+import Vuex from 'vuex'
+import DesignSystem from 'owncloud-design-system'
+
+import Avatar from 'web-runtime/src/components/Avatar.vue'
+
+const localVue = createLocalVue()
+localVue.use(Vuex)
+localVue.use(DesignSystem)
+
+const propsData = {
+  userName: 'admin',
+  userid: 'admin',
+  width: 24
+}
+
+global.fetch = jest.fn(() =>
+  Promise.resolve({
+    ok: false,
+    status: 404
+  })
+)
+
+const ocSpinner = 'oc-spinner-stub'
+const ocAvatar = 'oc-avatar-stub'
+
+describe('Avatar component', () => {
+  describe('when loading is set as true', () => {
+    it('should render oc-spinner but not oc-avatar', () => {
+      const wrapper = getShallowWrapper(true)
+      const spinner = wrapper.find(ocSpinner)
+      const avatar = wrapper.find(ocAvatar)
+
+      expect(avatar.exists()).toBeFalsy()
+      expect(spinner.exists()).toBeTruthy()
+      expect(spinner.attributes().style).toEqual(
+        `width: ${propsData.width}px; height: ${propsData.width}px;`
+      )
+    })
+  })
+
+  describe('when loading is set as false', () => {
+    let wrapper
+    beforeEach(() => {
+      wrapper = getShallowWrapper()
+    })
+
+    it('should render oc-avatar but not oc-spinner', () => {
+      const spinner = wrapper.find(ocSpinner)
+      const avatar = wrapper.find(ocAvatar)
+
+      expect(spinner.exists()).toBeFalsy()
+      expect(avatar.exists()).toBeTruthy()
+    })
+    it('should set props on oc-avatar component', () => {
+      const avatar = wrapper.find(ocAvatar)
+
+      expect(avatar.props().width).toEqual(propsData.width)
+      expect(avatar.props().userName).toEqual(propsData.userName)
+      expect(avatar.props().src).toEqual('')
+    })
+  })
+})
+
+function getShallowWrapper(loading = false) {
+  return shallowMount(Avatar, {
+    localVue,
+    store: createStore(),
+    propsData,
+    data() {
+      return {
+        loading
+      }
+    }
+  })
+}
+
+function createStore() {
+  return new Vuex.Store({
+    getters: {
+      getToken: jest.fn(() => 'GFwHKXdsMgoFwt'),
+      capabilities: jest.fn(() => ({
+        files_sharing: {
+          user: {
+            profile_picture: true
```

Comment:
Please check the `Avatar component` and have a test for `profile_picture` being both set to true and to false (and checking it causes the desired behavior)!

Reply:
Edit: Could also happen in an integration test I suppose. For the unit test it should be enough to spy & check that the `setUser` method is successfully being called on mounted